src/vsr.zig

```diff
@@ -347,37 +355,71 @@ pub const Header = extern struct {
 
     fn invalid_ping(self: *const Header) ?[]const u8 {
         assert(self.command == .ping);
+        if (self.checksum_body != checksum_body_empty) return "checksum_body != expected";
```

Comment:
Let's place that right next to `if (self.size != @sizeOf(Header))`: I think we always want both asserts together?

Reply:
:+1: Yeah, they are always accompanying each other.

(Fyi my motivation for putting them at the front was that most of the field validations are (roughly) ordered to mirror the `Header` struct layout).